prov/efa/src/efa_mr.c

```diff
@@ -928,8 +929,17 @@ static int efa_mr_reg_impl(struct efa_mr *efa_mr, uint64_t flags, void *attr)
 			shm_flags |= FI_HMEM_DEVICE_ONLY;
 		}
 
-		ret = fi_mr_regattr(efa_mr->domain->shm_domain, attr,
+		if (FI_VERSION_GE(efa_mr->domain->util_domain.fabric->fabric_fid.api_version, FI_VERSION(1, 19))) {
+			hmem_data = mr_attr->hmem_data;
+			mr_attr->hmem_data = efa_mr->peer.hmem_data;
```

Comment:
I do not understand why we will pass hmem_data from user. Don't you want to pass gdrcopy handle?

Reply:
> I think we need to add the OFI_GDRCOPY_HANDLE flag

Yes. It is done on L926 `shm_flags = efa_mr->peer.flags`

Reply:
On a second look. This is a good catch. We also need to respect the user flag. Will update.

Reply:
I see. A comment would help here.

Reply:
Thank you. Addressed.

Reply:
Sorry, i did not follow, where is the `OFI_GDRCOPY_HANDLE` being set?

Reply:
i see, makes sense